### ISSUE TITLE
Incorrect diagnostic for UserDict subclass constructor call: "Too many arguments" 

### ISSUE THREAD

**Issue by elprans** [python/mypy#4940]
Consider the following:

*test.py*:

```python
import collections


class Foo(collections.UserDict):
    @classmethod
    def from_dict(cls, dct: dict) -> 'Foo':
        return cls(dct)  # this line 7
```

running `mypy test.py` gives:

```
test.py:7: error: Too many arguments for "Foo"
```

```
$ mypy --version
mypy 0.590
```

**Comment by ilevkivskyi** [comment]
typeshed problem

**Comment by ilevkivskyi** [comment]
Just to clarify, all such problems (like missing `__init__` in stub for `UserDict`) should reported directly on https://github.com/python/typeshed
